Assets/MixedRealityToolkit.Tests/PlayModeTests/PrimitiveManipulatorTests.cs

```diff
@@ -0,0 +1,1000 @@
+            yield return leftHand.Show(new Vector3(-0.1f, -0.1f, 0.5f));
+            yield return null;
+
+            // One Handed
+            manipHandler.ManipulationType = PrimitiveManipulator.HandMovementType.OneHanded;
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(1, manipulationStartedCount);
+            Assert.AreEqual(0, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(1, manipulationStartedCount);
+            Assert.AreEqual(0, manipulationEndedCount);
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(1, manipulationStartedCount);
+            Assert.AreEqual(1, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(1, manipulationStartedCount);
+            Assert.AreEqual(1, manipulationEndedCount);
+
+            // Two Handed
+            manipHandler.ManipulationType = PrimitiveManipulator.HandMovementType.TwoHanded;
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(1, manipulationStartedCount);
+            Assert.AreEqual(1, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(2, manipulationStartedCount);
+            Assert.AreEqual(1, manipulationEndedCount);
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(2, manipulationStartedCount);
+            Assert.AreEqual(2, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(2, manipulationStartedCount);
+            Assert.AreEqual(2, manipulationEndedCount);
+
+            // One + Two Handed
+            manipHandler.ManipulationType = PrimitiveManipulator.HandMovementType.OneHanded | PrimitiveManipulator.HandMovementType.TwoHanded;
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(3, manipulationStartedCount);
+            Assert.AreEqual(2, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            Assert.AreEqual(3, manipulationStartedCount);
+            Assert.AreEqual(2, manipulationEndedCount);
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(3, manipulationStartedCount);
+            Assert.AreEqual(2, manipulationEndedCount);
+
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            Assert.AreEqual(3, manipulationStartedCount);
+            Assert.AreEqual(3, manipulationEndedCount);
```

Comment:
Now that the test has been already written, this is more of a consideration for next time, but in general having tests who state assertions depend on previous state (i.e. the fact that things start at 2 in this test case are because the previous "cases" already exist). This makes it hard to do things like delete things from earlier parts in the test (imagine we were to add a new test case in the middle after the first case and before the second case, we'd have to change up everything after it).

Where in reality, these really are separate test cases (i.e. PrimitiveManipulatorOneHandedEvents, PrimitiveManipulatorTwoHandedEvents, PrimitiveManipulatorOneAndTwoHandedEvents)

This is something that may not actually matter but when it does, it's something that you would wish that you had had structured differently.